internal/clientconn/conn.go

```diff
@@ -79,10 +79,6 @@ type newConnOpts struct {
 
 // newConn creates a new client connection for given net.Conn.
 func newConn(opts *newConnOpts) (*conn, error) {
-	if opts.handler == nil {
```

Comment:
Why is this deleted? Is this `opts.handler == nil` condition checked somewhere else?

Reply:
it's deleted bcz of linter error (cannot compare handler to nil)

Reply:
Well, `opts.handler` is an interface, and it can be nil. Please revert that change

Reply:
@AlekSi should it panic though? Or should it return an error instead?

Reply:
> Well, `opts.handler` is an interface, and it can be nil. Please revert that change

done

Reply:
> should it panic though? Or should it return an error instead?

It should not be possible for a user to do something that would trigger that case, so that's a programmer's error, and we "handle" it by panicking

Reply:
Ok! Sounds reasonable!